### ISSUE TITLE
A minor tweak, simplify

### ISSUE THREAD

**Issue by JustGregory-zz** [LonelyGayTiger/TheCorruptionOfKindness#1]
https://github.com/LonelyGayTiger/TheCorruptionOfKindness/blob/c70a2b4da46ddcaca5d8f84f88f3d31251d1624f/src/com/CorruptionOfKindness/Body/Bladder.java#L48-L62

I believe this can be simplified. I've seen this kind of methodology in more than one class, where checking if A + B is greater than C, but I'm using this as a basis for the comment.

Why not try putting the ```else``` body's ```this.urine += urine``` above the ```if``` statement? Either way, the urine must be added; then you only need to check if current urine is greater than or equal to max urine -- if it is, the character's gonna leak something...

And what happens while the character is asleep, entranced, hypnotized, or otherwise not consciously able to handle manual urination? Wetting the bed at night may be a problem for young children through to their pubescent years, and perhaps for a small percentage of adults and senior citizens as well; but for the most part as I understand it, the human brain adapts to not allow urination while asleep, except in critical situations.

Also in that case, it's not likely that the character will _only_ leak just down to the max urine stat; it is more likely an uncontrollable thing that will cause a variable percent of the total urine to run out -- it's variable, but let's say for simplicity's sake that it might be between 25-55 percent of the total urine. The subconscious part of the brain is the controller for involuntary (and probably voluntary, now that I think of it) urination, so it's gonna make sure the bladder has room for later urine additions.

Actually there might be a further flaw: even if the character takes in no food or drink, urine will continue to be added; in reality I think there should be a handler to administer translating food and drink to feces and urine, so that a certain period of time into starvation, less and less of the feces and urine will be added -- getting down to fractional degrees, but oh well.

I guess I would ask if this is still being developed, or has been abandoned. I have further ideas for the project but am offering a minor suggestion first. In many ways this could be added to, enhanced, and made part of a game or even educational software, with a few tweaks and additions. (Not that I want to create classes for each and every body organ, but I have further ideas for this project as a physics simulation, if you don't mind me borrowing and adapting for my shameless little project... )

**Comment by LonelyGayTiger** [comment]
Hey, thanks for letting me know. I just implemented the fix specifically for the urine class, tomorrow I will go and search out the rest of those errors. I am in fact still working on this. This is kind of a long haul project for me and I try to work on it whenever education or work don't get in the way, which ends up being most of the time. If you wanted to use part of the code or even just any of the classes at all, go right ahead! As long as you make it obvious which pieces of code originally came from me. I would also appreciate a link to whatever project it get's used for, though that's mostly because I just want to see what cool stuff you do with it. 

Thanks,
       Leo King

**Comment by JustGregory-zz** [comment]
Would a general attribution in the credits be fine?

But the thing is, I've got a project I was working on before I found this, a dead fork of the "Corruption of Champions" Flash project, that I had been translating from ActionScript to Java, painstakingly class by class, for a similar game framework. I'm sure that some of your and my works have overlapped without me knowing your work before. Until I sort out what is what, I would surely attribute your project here as having an influence on my own project (if I continue with it).

Right now it's just a working copy to tinker with, I'm not even making a Github project for it yet. For reasons I'm not going into it as sex-heavy as CoC or CoK may be; although that may be a natural option in the game.

Closing for now, as you have committed your changes that I'll take a deeper look into; but if you have further comment in this regard, I'll keep an ear open to this....
